### PR TITLE
Add PolyData intersection filter

### PR DESCRIPTION
**Pull request by whophil**
### Overview

Add `PolyData.intersection` filter ([vtkIntersectionPolyData](https://vtk.org/doc/nightly/html/classvtkIntersectionPolyDataFilter.html)).

### Details

I think this is a useful capability to wrap, and I also need it for my own uses (https://discourse.vtk.org/t/splitting-cells-on-a-polydata/5021/7)

<details>
<summary>code for demo</summary>

```python
import pyvista as pv

s1 = pv.Sphere(center=(0, 0, 0), radius=10)
s2 = pv.Sphere(center=(5, 0, 0), radius=10)

intersection, s1_new, s2_new = s1.intersection(s2)

p = pv.Plotter()
p.add_mesh(intersection, color='r')
p.add_mesh(s1, opacity=0.5)
p.add_mesh(s2, opacity=0.5)
p.show()
```
</details>

image

### CODE REVIEW

File: pyvista/core/filters.py [hunk 1]
```diff
@@ -2884,6 +2884,49 @@ def boolean_difference(poly_data, mesh, inplace=False):
...
+            The intersection line.
+
+        first_split: pyvista.PolyData
+            The first mesh split along the intersection. Returns the original first mesh
+            if `split_first` is False.
+
+        second_split: pyvista.PolyData
+            The second mesh split along the intersection. Returns the original second mesh
+            if `split_second` is False.
+        """
```
Comment (akaszynski) [inline pyvista/core/filters.py:2915]: An example here would be really helpful.  I'd just copy part of what you did in the test and place it here.

**Comment by whophil** [comment]
There is an issue with one of the rotation hypothesis tests (link [here](https://dev.azure.com/pyvista/PyVista/_build/results?buildId=4377&view=logs&j=d32c45aa-99ce-5d7c-7eaa-2fb21fc5cfb7&t=d676deb6-faaa-5297-b3bd-b90db128afa6&l=553)). Could be unrelated to these changes?

**Comment by akaszynski** [comment]
> There is an issue with one of the rotation hypothesis tests (link [here](https://dev.azure.com/pyvista/PyVista/_build/results?buildId=4377&view=logs&j=d32c45aa-99ce-5d7c-7eaa-2fb21fc5cfb7&t=d676deb6-faaa-5297-b3bd-b90db128afa6&l=553)). Could be unrelated to these changes?

Completely unrelated.  If this happens again I'll patch that CI test.

Thanks for adding the examples.  Approving and merging.

File: pyvista/core/filters.py [hunk 2]
```diff
@@ -2701,7 +2701,7 @@ def outline_corners(composite, factor=0.2, nested=False):
             controls the relative size of the corners to the length of the
             corresponding bounds
 
-        ested : bool, optional
```
Comment (akaszynski) [inline pyvista/core/filters.py:2704]: Good catch BTW.

**Review by akaszynski — APPROVED** [review]

**Pull request merged**